src/Raven.Server/ServerWide/ClusterStateMachine.cs

```diff
@@ -591,8 +595,29 @@ private void ExecutePutSubscriptionBatch(TransactionOperationContext context, Bl
                             $"Cannot set typed value of type {type} for database {database}, because it does not exist");
                     }
 
-                    var id = updateCommand.FindFreeId(context, index);
-                    updateCommand.Execute(context, items, id, record: null, _parent.CurrentState, out _);
+                    long minimumSubscriptionId = GetMinimumSubscriptionIdFor(database);
+
+                    long subscriptionId;
+
+                    if (updateCommand.SubscriptionId.HasValue)
+                    {
+                        if (updateCommand.SubscriptionId.Value < minimumSubscriptionId)
+                            minimumSubscriptionIdPerDatabase[database] = minimumSubscriptionId;
+
+                        subscriptionId = updateCommand.SubscriptionId.Value;
+                    }
+                    else
+                    {
+                        subscriptionId = index;
+                        if (subscriptionId >= minimumSubscriptionId)
+                        {
+                            subscriptionId = --minimumSubscriptionId;
```

Comment:
Why are you _reducing_ the `minimumSubscriptionId`?

Reply:
The `SubscriptionId` is generated using the current index and must be unique.
When creating a new subscription there is no issue but when importing using Smuggler we use a batch command.
This is the only way to keep the id unique without collisions when we save the next subscription id.